torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -175,6 +175,17 @@ std::tuple<Tensor, Tensor> _euclidean_dist_backward(const Tensor & grad, const T
             x2 * ratio.sum(-2, false).unsqueeze(-1) - ratio.transpose(-2, -1).matmul(x1)};
 }
 
+Tensor _to_copy_backward(const Tensor& grad, const c10::TensorOptions& input_options) {
+  return _to_copy(
+      grad,
+      input_options.dtype().toScalarType(),
+      input_options.layout(),
+      input_options.device(),
+      input_options.pinned_memory(),
+      /*non_blocking=*/false,
```

Comment:
Why isn't the non-blocking propagated here?

Reply:
It's not propagated in copy backward: https://github.com/pytorch/pytorch/blob/48af9de92ffae89618a22291738189e3110d14ed/torch/csrc/autograd/functions/tensor.cpp#L34. Since _to_copy is replacing copy backward for `at::to`, I wanted to replicate the behavior.

Another way to think about it is: It doesn't matter in the backward pass if `_to_copy` blocks or not; it only matters to the user that it blocks in the forward pass.

Reply:
For CPU -> CUDA it doesn't matter indeed. Because the copy the other way around cannot be non-blocking.
But for CUDA -> CUDA copy maybe we should do that actually?

cc @mcarilli ?